Source/Core/Core/IOS/ES/Formats.cpp

```diff
@@ -600,32 +605,34 @@ bool SharedContentMap::WriteEntries() const
          HLE::FS::ResultCode::Success;
 }
 
-static std::pair<u32, u64> ReadUidSysEntry(const HLE::FS::FileHandle& file)
+static std::pair<u32, u64> ReadUidSysEntry(HLE::FSDevice& fs, u64 fd, u64* ticks)
```

Comment:
Most functions use `std::shared_ptr`, is there a reason to switch to a  `non-const reference` rather a pointer. It's less obvious that's it's being modified and also violates Dolphin's coding style.

Reply:
There's no point to passing a shared_ptr by copy considering this function doesn't share ownership. It just needs a `fs` instance that actually exists and pointer semantics aren't useful here since `fs` cannot be null (and *must not* be null) and it's already clear that the instance is being modified in some way.

The guideline about using pointers rather than references is meant for situations where the value that is modified is e.g. an integer or a string, when it's otherwise unclear that the value is modified. In this case, since a device is an opaque handle anyway, what additional information would `ReadUidSysEntry(fs.get(), fd, &m_ticks);` give you over `ReadUidSysEntry(*fs, fd, &m_ticks);`? Absolutely nothing, but with a pointer you open the door to ReadUidSysEntry being misused and called with nullptr when a reference tells you at the type system level that the function requires a valid fs.

Reply:
These explanations sound fine to me.